### PR TITLE
Add subctl installation instructions

### PR DESCRIPTION
**Pull request by skitt**
Signed-off-by: Stephen Kitt <skitt@redhat.com>

### CODE REVIEW

**Comment by mangelajo** [comment]
This change is [Reviewable](https://reviewable.io/reviews/submariner-io/submariner/246)

File: README.md [hunk 1]
```diff
@@ -90,7 +94,46 @@ An example of three clusters configured to use with Submariner would look like t
...
+
+## Installation using subctl
+
+`subctl` is a Submariner-specific tool developed to simplify Submariner installation. It drives
+[the operator](#installation-using-operator). You can see it in action in
+[this demo](https://youtu.be/cInmBXuZsU8).
+
+### Download
+
+`subctl` is maintained in the operator project; binaries are available for download from
```
Comment (tpantelis) [inline README.md]:
```suggestion
`subctl` is maintained in the operator project and its binaries are available for download from
```

File: README.md [hunk 2]
```diff
@@ -90,7 +94,46 @@ An example of three clusters configured to use with Submariner would look like t
...
+[the operator releases](https://github.com/submariner-io/submariner-operator/releases).
+Download the latest binary for your operating system and architecture, and install it on your path;
```
Comment (tpantelis) [inline README.md]:
```suggestion
Download the latest binary for your operating system and architecture and install it on your path.
```

File: README.md [hunk 3]
```diff
@@ -90,7 +94,46 @@ An example of three clusters configured to use with Submariner would look like t
...
+it has no external dependencies.
```
Comment (tpantelis) [inline README.md]:
```suggestion
It has no external dependencies.
```

File: README.md [hunk 4]
```diff
@@ -90,7 +94,46 @@ An example of three clusters configured to use with Submariner would look like t
...
+
+### Broker
+
+To set the broker up (on a specific cluster or one of the multi-cluster member clusters), run
```
Comment (tpantelis) [inline README.md]:
```suggestion
To deploy only the broker in a cluster, run
```

File: README.md [hunk 5]
```diff
@@ -90,7 +94,46 @@ An example of three clusters configured to use with Submariner would look like t
...
+
+    subctl deploy-broker --kubeconfig /path/to/your/config --no-dataplane
+
+`/path/to/your/config` must correspond to the `kubeconfig` for your cluster.
```
Comment (tpantelis) [inline README.md]:
```suggestion
`/path/to/your/config` must correspond to the `kubeconfig` of the target cluster.
```

File: README.md [hunk 6]
```diff
@@ -90,7 +94,46 @@ An example of three clusters configured to use with Submariner would look like t
...
+
+This will deploy the broker on the target cluster, and save a file, `broker-info.subm`, that will
```
Comment (tpantelis) [inline README.md]:
```suggestion
This command also saves a file, `broker-info.subm`, that will
```

Where is this file saved?

File: README.md [hunk 7]
```diff
@@ -90,7 +94,46 @@ An example of three clusters configured to use with Submariner would look like t
...
+be used to join the multi-cluster.
```
Comment (tpantelis) [inline README.md]:
```suggestion
be used to join other clusters to the broker.
```

File: README.md [hunk 8]
```diff
@@ -90,7 +94,46 @@ An example of three clusters configured to use with Submariner would look like t
...
+
+### Engine and route agent
+
+To install the engine and route agent, run
+
+    subctl join --kubeconfig /path/to/your/config broker-info.subm
+
+Again, `/path/to/your/config` must correspond to the `kubeconfig` for your cluster; it will vary
```
Comment (tpantelis) [inline README.md]:
```suggestion
`/path/to/your/config` must correspond to the `kubeconfig` of the joining cluster.
`broker-info.subm` is the file obtained from the broker deployment and is used to join the cluster to that broker.
```

File: README.md [hunk 9]
```diff
@@ -90,7 +94,46 @@ An example of three clusters configured to use with Submariner would look like t
...
+from one cluster to another.
```
Comment (tpantelis) [inline README.md]: Remove this line.

File: README.md [hunk 10]
```diff
@@ -90,7 +94,46 @@ An example of three clusters configured to use with Submariner would look like t
...
+
+This will ask for any missing information, in particular a cluster identifier (a unique name for
```
Comment (tpantelis) [inline README.md]:
```suggestion
This command will ask for any missing information, in particular a unique identifier/name for
```

File: README.md [hunk 11]
```diff
@@ -90,7 +94,46 @@ An example of three clusters configured to use with Submariner would look like t
...
+the cluster), and the node to use as a gateway.
```
Comment (tpantelis) [inline README.md]:
```suggestion
the cluster and the cluster node to use as a gateway.
```

File: README.md [hunk 12]
```diff
@@ -90,7 +94,46 @@ An example of three clusters configured to use with Submariner would look like t
...
+It will then install [the Submariner operator](#installation-using-operator) and configure it to
+deploy Submariner and join the clusters configured using the broker above.
```
Comment (tpantelis) [inline README.md]:
```suggestion
deploy the Submariner components and join the cluster to the broker.
```

File: README.md (hunk 2 above)
Comment (skitt) [reply README.md]: Losing the comma makes the previous “and” less clear — the two “and”s are different...

File: README.md (hunk 3 above)
Comment (skitt) [reply README.md]: This is part of the sentence on the previous line.

File: README.md (hunk 2 above)
Comment (tpantelis) [reply README.md]: How about this:

"Download the latest binary for your operating system and architecture, then install it on your path, as the binary has no external dependencies."

Comment (skitt) [reply README.md]: Right, “then” works better. There’s no connection between the path and the external dependencies, so I changed it to

> Download the latest binary for your operating system and architecture, then install it on your
path. The binary has no external dependencies so you won’t need anything else.

**Pull request merged**